File: Utils/Classes/Mensagem_comum.php
<?php
    include_once "C:/xampp/htdocs/WideLancer_Artefato/Utils/DatabaseFunctions.php";
    include_once "C:/xampp/htdocs/WideLancer_Artefato/Utils/Classes/Mensagem.php";
    
class MensagemComum extends Mensagem{
    private $texto;
    private $imagem;

    //construtores
    public function __construct($msg_id, $texto, $imagem, $horario, $chat_id, $usuario_id){
        parent::__construct($msg_id, $horario, $chat_id, $usuario_id, "mensagem_comum");
        $this->texto = $texto;
        $this->imagem = $imagem;
    }

    //getters e setters
        public function getTexto() {
            return $this->texto;
        }

        public function setTexto($texto) {
            $this->texto = $texto;
        }

        public function getImagem() {
            return $this->imagem;
        }

        public function setImagem($imagem) {
            $this->imagem = $imagem;
        }

        public function getMensagemId() {
            return parent::getId();
        }


    //metodos publicos 
    public function cadastrar(){
        $bd = ConectarSQL();
        $sql = "CALL cadastrar_msg_comum(?, ?, ?, ?)";

        $query = $bd->prepare($sql);
        echo $this->chat_id;
        $query->bind_param("ssii", $this->texto, $this->imagem, $this->chat_id, $this->usuario_id);
        $cadastrada = $query->execute();

        $bd->close();
        return $cadastrada;
    }

    //metodos estaticos
}

?>

File: src/main/java/com/wideLancer/models/MensagemComum.java
package com.wideLancer.models;

import com.wideLancer.abstractions.Mensagem;

public class MensagemComum extends Mensagem{

    private int id;
    private String texto;
    private String imagem;
    private int emissor;
    private int chat;

    @Override
    public String getTipo(){
        return "MensagemComum";
    }

    public int getId(){
        return this.id;
    }

    public int getEmissor(){
        return this.emissor;
    }

    public int getChat(){
        return this.chat;
    }

    public String getImagem(){
        return this.imagem;
    }

    public String getTexto(){
        return this.texto;
    }

    public void setId(int id){
        this.id = id;
    }

    public void setEmissor(int emissor){
        this.emissor = emissor;
    }

    public void setChat(int chat){
        this.chat = chat;
    }

    public void setImagem(String imagem){
        this.imagem = imagem;
    }

    public void setTexto(String texto){
        this.texto = texto;
    }
}
